src/client/sandbox/event/index.ts

```diff
@@ -171,33 +172,45 @@ export default class EventSandbox extends SandboxBase {
     attach (window: Window & typeof globalThis): void {
         super.attach(window);
 
-        window.HTMLInputElement.prototype.setSelectionRange    = this._overriddenMethods.setSelectionRange;
-        window.HTMLTextAreaElement.prototype.setSelectionRange = this._overriddenMethods.setSelectionRange;
+        overrideFunction(window.HTMLInputElement.prototype, 'setSelectionRange', this._overriddenMethods.setSelectionRange);
+        overrideFunction(window.HTMLTextAreaElement.prototype, 'setSelectionRange', this._overriddenMethods.setSelectionRange);
 
         if (isIE11) {
-            window.Window.prototype.dispatchEvent      = this._overriddenMethods.dispatchEvent;
-            window.Document.prototype.dispatchEvent    = this._overriddenMethods.dispatchEvent;
-            window.HTMLElement.prototype.dispatchEvent = this._overriddenMethods.dispatchEvent;
-            window.SVGElement.prototype.dispatchEvent  = this._overriddenMethods.dispatchEvent;
+            overrideFunction(window.Window.prototype, 'dispatchEvent', this._overriddenMethods.dispatchEvent);
+            overrideFunction(window.Document.prototype, 'dispatchEvent', this._overriddenMethods.dispatchEvent);
+            overrideFunction(window.HTMLElement.prototype, 'dispatchEvent', this._overriddenMethods.dispatchEvent);
+            overrideFunction(window.SVGElement.prototype, 'dispatchEvent', this._overriddenMethods.dispatchEvent);
         }
-        else
-            window.EventTarget.prototype.dispatchEvent = this._overriddenMethods.dispatchEvent;
+        else {
+            overrideFunction(window.EventTarget.prototype, 'dispatchEvent', this._overriddenMethods.dispatchEvent);
+        }
+
+        overrideFunction(window.HTMLElement.prototype, 'focus', this._overriddenMethods.focus);
+        overrideFunction(window.HTMLElement.prototype, 'blur', this._overriddenMethods.blur);
+        overrideFunction(window.HTMLElement.prototype, 'click', this._overriddenMethods.click);
 
-        window.HTMLElement.prototype.focus    = this._overriddenMethods.focus;
-        window.HTMLElement.prototype.blur     = this._overriddenMethods.blur;
-        window.HTMLElement.prototype.click    = this._overriddenMethods.click;
-        window.Event.prototype.preventDefault = this._overriddenMethods.preventDefault;
+        overrideFunction(window.Event.prototype, 'preventDefault', this._overriddenMethods.preventDefault);
 
         // @ts-ignore Window constructor has no the focus method
         window.Window.focus = this._overriddenMethods.focus;
+
+        // NOTE:
+        // we cannot use 'overrideFunction' here since the focus method may not exist
+        // @ts-ignore Window constructor has no the focus method
+        overrideStringRepresentation(window.Window.focus, nativeMethods.focus);
```

Comment:
Add the necessary checks into the `overrideFunction` and use it here.

Remove the `overrideStringRepresentation` function and use the combination of `overrideFunctionName` and `overrdieToString` instead.

Reply:
I assume that we shouldn't assign our wrapper to a native function when it is missing (e.g. `window.Window.focus`, which is undefined in IE11). It would be much better to flag these cases somehow (which I did by combining plain assignment and using `overrideStringRepresentation` function). We need to ensure that we override only those native functions that exist.

Reply:
In this case we can omit any flags because it clear behavior - override only existing functions.
Also, in general case the flag is the worst way to pass parameters.
See https://github.com/ryanmcdermott/clean-code-javascript#dont-use-flags-as-function-parameters

Reply:
After a more in-depth discussion with @miherlosev and @LavrovArtem, we decided to use `overrideStringRepresentation` in cases like this. It's temporarily and later it'll be implemented properly through `overrideFunction`.